File: erettsegi/kozep/2022_okt/java/Kockak.java
import module java.base;

public class Kockak {

    public static void main(String[] args) {
        var random = new Random();
        var alkalmakSzama = Integer.parseInt(System.console().readLine("Hány alkalommal legyen feldobás? "));
        var korok = new ArrayList<Kor>();

        while(korok.size() != alkalmakSzama) {
            var kor = kortGeneral(random);

            if(!korok.contains(kor)) {
                korok.add(kor);
            }
        }

        var anniNyertKorok = 0;
        var panniNyertKorok = 0;

        for(var kor : korok) {
            var anniNyert = kor.osszeg < 10;

            System.out.println("Dobás: " + kor.elso + " + " + kor.masodik + " + " + kor.harmadik + " = " + kor.osszeg + ", nyert: " + (anniNyert ? "Anni" : "Panni"));

            if(anniNyert) {
                ++anniNyertKorok;
            }else{
                ++panniNyertKorok;
            }
        }

        System.out.println("A játékok során " + anniNyertKorok + " alkalommal nyert Anni, " + panniNyertKorok + " alkalommal pedig Panni");
    }

    static Kor kortGeneral(Random random) {
        var elso = dobastGeneral(random);
        var masodik = dobastGeneral(random);
        var harmadik = dobastGeneral(random);

        return new Kor(elso, masodik, harmadik, elso + masodik + harmadik);
    }

    static int dobastGeneral(Random random) {
        return 1 + random.nextInt(6);
    }

    record Kor(int elso, int masodik, int harmadik, int osszeg) {}
}

File: erettsegi/kozep/2022_okt/python/kockak.py
import random

alkalmak_szama = int(input('Hány alkalommal legyen feldobás? '))
korok = []

while len(korok) != alkalmak_szama:
    elso = random.randint(1, 6)
    masodik = random.randint(1, 6)
    harmadik = random.randint(1, 6)

    dobasok = [ elso, masodik, harmadik, elso + masodik + harmadik ]

    if not dobasok in korok:
        korok.append(dobasok)

anni_nyert_korok = 0
panni_nyert_korok = 0

for kor in korok:
    anni_nyert = kor[3] < 10

    print(f'Dobás: {kor[0]} + {kor[1]} + {kor[2]} = {kor[3]}, nyert: {"Anni" if kor[3] < 10 else "Panni"}')

    if anni_nyert:
        anni_nyert_korok += 1
    else:
        panni_nyert_korok += 1

print(f'A játékok során {anni_nyert_korok} alkalommal nyert Anni, {panni_nyert_korok} alkalommal pedig Panni')
